realm/realm-library/src/main/cpp/util_sync.cpp

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util.hpp"
+#include "util_sync.hpp"
+
+// Must match OSJNIBsonProtocol.VALUE
+static const std::string VALUE("value");
+
+using namespace realm::bson;
+
+Bson jstring_to_bson(JNIEnv* env, jstring arg) {
+    JStringAccessor args_json(env, arg);
+    BsonDocument document(parse(args_json));
+    return document[VALUE];
```

Comment:
Is this the best approach? Why not return the `BsonDocument` instance instead? You could have an additional method that provides a `Bson` too. There are multiple methods receiving a document in `remote_mongo_collection.hpp`, e.g.

`void count(const bson::BsonDocument& filter_bson ...`

Reply:
The reason for wrapping them is that I cannot find a proper way to parse non-document values, which is also argument/response to a number of the C++ calls. The `BsonDocument` is however itself a `BsonValue` so there should be no problem embedding these in the convention (besides the problem that current C++ parser yields different types than the original document and I cannot pass `BsonArray`...but I am working on resolving those issues)

Reply:
The idea is also to provide a wrapper for easier wrapping/unwrapping correctly typed values, as in https://github.com/realm/realm-java/pull/6810/files#diff-84db7612944e18ff25865af1cb6daa17